### DSBA M1
## Econometrics
Prof. Roberto Renò - ESSEC Business School 

## Homework 1

In [99]:
import numpy as np
import pandas as pd

In [100]:
data_in = ""
data = pd.read_csv(data_in + "bwages.csv")

In [101]:
data.head()

,WAGE,LNWAGE,EDUC,EXPER,LNEXPER,LNEDUC,MALE
0,7.780208,2.051583,1,23,3.178054,0.0,1
1,4.818505,1.572464,1,15,2.772589,0.0,0
2,10.563645,2.357418,1,31,3.465736,0.0,1
3,7.042429,1.951953,1,32,3.496508,0.0,1
4,7.887521,2.065282,1,9,2.302585,0.0,1


## Explaining wages

We model individual wages using a cross-sectional data set of 1472 observations. The dataset is called BWAGES and the variables we use are:

• wage: hourly wage rate in euros per hous
• male: 1 if male, 0 if female
• educ: education level, 1 = primary school, 2 = lower vocational training, 3 = intermediate level, 4 = higher vocational training, 5 = university level
• exper: experience in years

## Questions 

Part A

1. Compute standard descriptive statistics for the variables involved in this study.
2. Prepare extra columns in the data dataframe with dummy variables: educ1=1 if primary school, 0 otherwise; educ2=1 if lower vocational training, 0 otherwise, and so on.
3. Estimate the following model by OLS: wage = β0 + β1male + β2educ1 + β3educ2 + β3educ3 + β4educ4 + β5exper + ε, and interpret the estimated coefficients.
4. Why is there no educ5 in the model? Explain.
5. What is the expected salary for a man with university level education?
6. Using standard t-tests, are the explanatory variables significant?
7. Is there a significant salary difference between men and woman according to the above model?
8. Add squared experience as an additional variable in your model, estimate it again, and show the new results.
9. What is the impact of experience on wage? Write a short comment based on the statistical significance of the above estimate.

Part B

The above model did not have any interaction effect between gender and education.

10. Write down a model that explicitly incorporates an interaction effect between gender and education.
11. Estimate this new model using OLS and interpret the new coefficients.
12. Test for differences between men and woman in this general model.
13. Demonstrate empirically that the sum of squared residuals (SSR) for this general model is equal to the SSR of the model explaining wage by the educ dummies and the experience variables fitted to the sample of men + the SSR of the same model fitted to the sample of woman.

Part C

14. Test for the heteroskedasticity of the residuals.
15. Perform generalized least squares if homoskedasticity is rejected.
16. Compute White standard errors for the model in point 2 and compare them with the OLS standard errors found in question 2. What has changed?
17. Compute also HC1, HC2, HC3: what are the differences?
18. How much is the correlation between estimated residuals and exper? Explain.
19. Display the the distribution of the residuals ε in the model in point 2. Test its normality.
20. If the previous question rejects, what do you think are the consequences for the tests performed in this study? Explain

## Part 1
## Question 1

In [102]:
data.describe

<bound method NDFrame.describe of            WAGE    LNWAGE  EDUC  EXPER   LNEXPER    LNEDUC  MALE
0      7.780208  2.051583     1     23  3.178054  0.000000     1
1      4.818505  1.572464     1     15  2.772589  0.000000     0
2     10.563645  2.357418     1     31  3.465736  0.000000     1
3      7.042429  1.951953     1     32  3.496508  0.000000     1
4      7.887521  2.065282     1      9  2.302585  0.000000     1
...         ...       ...   ...    ...       ...       ...   ...
1467  11.119626  2.408712     5      1  0.693147  1.609438     1
1468   5.453375  1.696235     5     14  2.708050  1.609438     0
1469  11.085489  2.405637     5     19  2.995732  1.609438     0
1470  15.649843  2.750461     5     15  2.772589  1.609438     0
1471  15.023850  2.709639     5     24  3.218876  1.609438     1

[1472 rows x 7 columns]>

## Question 2

In [103]:
print(data.columns)

Index(['WAGE', 'LNWAGE', 'EDUC', 'EXPER', 'LNEXPER', 'LNEDUC', 'MALE'], dtype='str')


In [104]:
data['EDUC1'] = (data['EDUC'] == 1).astype(int)
data['EDUC2'] = (data['EDUC'] == 2).astype(int)
data['EDUC3'] = (data['EDUC'] == 3).astype(int)
data['EDUC4'] = (data['EDUC'] == 4).astype(int)
data['EDUC5'] = (data['EDUC'] == 5).astype(int)

data.head()

,WAGE,LNWAGE,EDUC,EXPER,LNEXPER,LNEDUC,MALE,EDUC1,EDUC2,EDUC3,EDUC4,EDUC5
0,7.780208,2.051583,1,23,3.178054,0.0,1,1,0,0,0,0
1,4.818505,1.572464,1,15,2.772589,0.0,0,1,0,0,0,0
2,10.563645,2.357418,1,31,3.465736,0.0,1,1,0,0,0,0
3,7.042429,1.951953,1,32,3.496508,0.0,1,1,0,0,0,0
4,7.887521,2.065282,1,9,2.302585,0.0,1,1,0,0,0,0


## Question 3

In [105]:
import statsmodels.api as sm

X = data[['MALE', 'EDUC1', 'EDUC2', 'EDUC3', 'EDUC4', 'EXPER']]
X = sm.add_constant(X)
y = data['WAGE']

model = sm.OLS(y, X).fit()

print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                   WAGE   R-squared:                       0.367
Model:                            OLS   Adj. R-squared:                  0.364
Method:                 Least Squares   F-statistic:                     141.5
Date:                Mon, 27 Apr 2026   Prob (F-statistic):          1.29e-141
Time:                        09:33:42   Log-Likelihood:                -3949.4
No. Observations:                1472   AIC:                             7913.
Df Residuals:                    1465   BIC:                             7950.
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         10.4010      0.260     40.012      0.0

## Question 4

### At first, we tried running the model with all five education dummies, but we quickly realized why educ5 has to be left out. It’s because of the Dummy Variable Trap and the issue of perfect multicollinearity. (Message from Python after computed with educ5)

### Our model already includes an intercept (the constant term β 0). 
### By definition, every single person in our dataset belongs to exactly one of the five education levels. This means that if you add up educ1 + educ2 + educ3 + educ4 + educ5, the sum is always 1 for every observation.

### Since the constant in our regression is also a column of 1s, having all five dummies creates a situation where one variable is a perfect linear combination of the others. Mathematically, this makes the X′X matrix 'singular' (non-invertible), and the OLS estimator can't be calculated properly.

### By dropping educ5, we turn educ5 into our base category (or reference group). Now, the intercept β 0 represents the average wage for someone with a university degree (educ5 group), and the coefficients for educ1 through educ4 tell us exactly how much (more or less) those groups earn compared to that university-level baseline. If we kept all five, we wouldn't have a reference point to compare them against

## Question 5

### Wage = 10.4010+(1.3009×1)+(0.1907×exper)
### Wage = 11.7019+0.1907×exper

### Expected Salary for a man with univ level education si 11,70€/h, and + 0,19€ for each year of experience

In [106]:
wage_base_male_univ = model.params['const'] + model.params['MALE']
print(f"Expected wage : {wage_base_male_univ:.4f} €/h")

Expected wage : 11.7019 €/h


## Question 6

### Based on the p-values in the P>∣t∣ column in Question 3, all explanatory variables are statistically significant at the 5% level, as all p-values are below 0.05 (all at 0.00). This means that gender, all levels of education (relative to university), and experience each have a confirmed statistical impact on the hourly wage.

## Question 7

Based on the regression results, the answer is yes. To determine this, we have to specifically look at the male variable.

First, if you look at the coefficient for male (which is 1.3009 in my output), it’s positive. This tells us that, on average, men earn about 1.30€ more per hour than women. Here, this condition means that this gap exists even when comparing a man and a woman who have the exact same level of education and the same years of experience.

Second, to see if this gap is 'real' or just due to random noise in our specific sample of 1472 people, I checked the p-value (the P>∣t∣ column). The p-value for male is 0.000. Since this is way below the standard significance threshold of 0.05 (5%), we reject the null hypothesis (H0:β male=0).

The probability that this 1.30€ difference is just a coincidence is basically zero. So, the model proves there is a statistically significant 'gender premium' in this dataset, favoring men.

## Question 8

In [110]:
from ipaddress import summarize_address_range
data['EXPER2'] = data['EXPER']**2

X = data[['MALE', 'EDUC1', 'EDUC2', 'EDUC3', 'EDUC4', 'EXPER', 'EXPER2']]
X = sm.add_constant(X)
Y = data['WAGE']

model = sm.OLS(Y,X).fit()

print(model.summary())


                            OLS Regression Results                            
Dep. Variable:                   WAGE   R-squared:                       0.380
Model:                            OLS   Adj. R-squared:                  0.377
Method:                 Least Squares   F-statistic:                     128.3
Date:                Mon, 27 Apr 2026   Prob (F-statistic):          3.36e-147
Time:                        09:43:19   Log-Likelihood:                -3933.8
No. Observations:                1472   AIC:                             7884.
Df Residuals:                    1464   BIC:                             7926.
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          9.3200      0.322     28.956      0.0

## Question 9

The impact of experience on wages is now non-linear, and the results are highly significant (with p-values of 0.000).

Specifically, the positive coefficient for exper (0.4859) and the negative coefficient for exper2 (−0.0075) indicate a concave relationship. This means that wages increase with experience, but they do so at a decreasing rate.

Statistically, this is very important: the 'return' on an extra year of experience is high at the beginning of a career, but it slowly diminishes over time. For example, the gain in wage for moving from 0 to 1 year of experience is much larger than the gain for moving from 30 to 31 years. 